masim.c

```diff
@@ -14,6 +14,16 @@
 #include "misc.h"
 #include "masim.h"
 
+// hugepage definitions
+#define PROTECTION (PROT_READ | PROT_WRITE)
```

Comment:
Let's use `/* */` style comment consistently.  Or, we could simply remove the comment by renaming the macro to a more self-explaining name, e.g., `HUGETLB_PROTECTION`.  I'd prefer the latter option.